components/filter/test/filter.visual-diff.js

```diff
@@ -107,10 +107,16 @@ describe('d2l-filter', () => {
 				it(type, async function() {
 					const selector = `#single-set-${type}`;
 					await open(page, selector);
-					await page.$eval(selector, async(filter) => {
-						filter.shadowRoot.querySelector('d2l-selection-select-all').shadowRoot.querySelector('d2l-input-checkbox').shadowRoot.querySelector('input').click();
+					await page.$eval(selector, async(filter, type) => {
+						const checkbox = filter.shadowRoot.querySelector('d2l-selection-select-all').shadowRoot.querySelector('d2l-input-checkbox');
+						if ((type === 'press-unselect-all' && checkbox.checked)
+							|| (type === 'press-select-all' && !checkbox.checked)
+						) {
+							// NOT visual-diff retry
+							checkbox.shadowRoot.querySelector('input').click();
```

Comment:
The checkbox was being re-clicked each retry, so this changes it to only click the first time

Reply:
I'm wondering if there might be a more general solution to "only run on first try" - if the window is persistent, then could we set a boolean flag on the window, something like `setupComplete`? Might make sense for it to be per-test, so like:

```js
// In a before block for the whole test:
window.setupComplete = {};

// In an individual test:
if (!window.setupComplete[this.test.fullTitle()] {
  // Do setup, then:
  window.setupComplete[this.test.fullTitle()] = true;
}

Reply:
Not sure whether this is worth the effort ☝️ ... Looking at some of the other changes, it seems like what we really need is cleanup after each test, to reset it to its initial state - or even better, starting fresh each time.

Reply:
Yup that could work, though agreed that it might not be worth the effort - the tests need an overhaul.